thealgorithms/other/anagrams.py

```diff
@@ -1,30 +1,34 @@
 from __future__ import print_function
 import collections, pprint, time, os
 
-start_time = time.time()
-print('creating word list...')
-path = os.path.split(os.path.realpath(__file__))
-with open(path[0] + '/words') as f:
-    word_list = sorted(list(set([word.strip().lower() for word in f])))
-
 def signature(word):
     return ''.join(sorted(word))
 
-word_bysig = collections.defaultdict(list)
-for word in word_list:
-    word_bysig[signature(word)].append(word)
-
-def anagram(myword):
+def anagram(word_bysig, signature, myword):
     return word_bysig[signature(myword)]
 
-print('finding anagrams...')
-all_anagrams = {word: anagram(word)
-                for word in word_list if len(anagram(word)) > 1}
+def main():
+    start_time = time.time()
+    print('creating word list...')
+    path = os.path.split(os.path.realpath(__file__))
+    with open(path[0] + '/words') as f:
+        word_list = sorted(list(set([word.strip().lower() for word in f])))
+
+    word_bysig = collections.defaultdict(list)
+    for word in word_list:
+        word_bysig[signature(word)].append(word)
+
+    print('finding anagrams...')
+    all_anagrams = {word: anagram(word)
+                    for word in word_list if len(anagram(word)) > 1}
```

Comment:
Actually, now that I look for it, you added args for anagram but call them here as in the original code.